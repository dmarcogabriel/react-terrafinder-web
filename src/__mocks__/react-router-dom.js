const originalModule = jest.requireActual('react-router-dom');

module.exports = {
  ...originalModule,
  useHistory: () => ({
    history: { push: jest.fn() },
  }),
};
