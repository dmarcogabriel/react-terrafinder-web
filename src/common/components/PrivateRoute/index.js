import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { useUser } from 'hooks/useUser';

export default function PrivateRoute({ path, children }) {
  const { currentUser } = useUser();

  // todo: prev route when unlogged

  if (currentUser && currentUser.token)
    return <Route path={path}>{children}</Route>;
  return <Redirect to="/login" />;
}
