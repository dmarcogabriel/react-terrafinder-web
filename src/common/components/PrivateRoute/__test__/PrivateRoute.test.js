import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
// import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import PrivateRoute from '..';

const Component = () => (
  <Router>
    <PrivateRoute />
  </Router>
);

jest.mock('hooks/useUser', () => ({
  useUser: () => ({
    currentUser: {
      token: 'test',
    },
  }),
}));

describe('<PrivateRoute />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Component />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<Component />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
