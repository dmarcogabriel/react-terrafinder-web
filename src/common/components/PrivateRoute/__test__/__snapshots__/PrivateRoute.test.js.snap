// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivateRoute /> matches snapshot 1`] = `null`;
